### ISSUE TITLE
Vermont shields and banner modifiers

### ISSUE THREAD

**Issue by ahallification** [ZeLonewolf/openstreetmap-americana#268]
Vermont is not nearly as complex as other states (Hi Ohio!), but there is some small nuance not currently captured by americana. Vermont has one state highway system and adopted their highway shield in the mid '90s. The default VT shield has already been created for americana. However, when the State of Vermont rolled out their highway shields, they only applied them to state maintained highways. The remaining numbered highways are town maintained, and are represented by the default state route shields, which are the default black and white ellipse. 

To capture this nuance, editing in osm has already been completed as follows:

1. For state maintained highways, no changes were made. The relationships still carry `network=US:VT`
2. For town maintained highways, the relationships were retagged with a new `network=US:VT:Town`. 
3. For highways that are split between state and town maintenance, the original relationship was split into two, with one relationship marked `name=VT xx (state)` and `name=VT xx (town)`. The town relationship was retagged with the new `network=US:VT:Town`.
4. There are also a couple bannered routes for "Alternate". These relationships were newly created and tagged `network=US:VT:Alternate`.

Since these are new edits, they are not represented in the current build (circa December 2021). The build will need to be updated as part of any changes to address this issue.

Example: [Junction of VT 22A and VT 73 shows the town and state maintained shields of VT 73](https://www.mapillary.com/app/?pKey=4361714303880587&mapStyle=esri&lat=43.804411522411&lng=-73.304979958851&z=17&focus=photo)

**Comment by ZeLonewolf** [comment]
#214 will fix the data issue.

**Comment by ZeLonewolf** [comment]
I've resolved #214, please let me know if/when a newer data update is needed to resolve this.

**Comment by ahallification** [comment]
I believe the data has been pulled correctly. This probably needs some magical code written followed by one of those pull request thingamabobbers. Though, with my non-software development background, I’m not entirely certain. I’m happy to advise, discuss, be coached, etc. whatever it takes to make my small contribution live in the demo.

**Comment by ZeLonewolf** [comment]
Step 1 is to head over to the [map](https://zelonewolf.github.io/openstreetmap-americana/#7/43.961/-72.139) and figure out if the shields are displaying as expected.  If they aren't, step 2 is figure out if the roads in question are tagged with the right tagging scheme.  Once that's determined, then we can figure out about thingamabobbers.

**Comment by ahallification** [comment]
All VT highways display the same shield currently. I’m pretty sure the data in the demo is up to date (tagging is correct), as I have recent edits that are now reflected.

So `network=US:VT` Is properly rendered, whereas the newer `network=US:VT:Town` still needs to be differentiated programmatically for rendering

**Comment by ZeLonewolf** [comment]
I can confirm that US:VT:Town is not currently differentiated.  What should those look like?

**Comment by zekefarwell** [comment]
Town maintained sections of Vermont State Routes have standard black and white oval shields as shown in this list:
https://en.wikipedia.org/wiki/List_of_state_highways_in_Vermont#Vermont_routes
I've opened a PR to add support for `US:VT:Town`, but the value is not yet in the tiles as far as I can tell.

**Comment by ahallification** [comment]
> There are also a couple bannered routes for "Alternate". These relationships were newly created and tagged `network=US:VT:Alternate`.

FYI, looks like a minor part of this got lost in the mix and I've only recently realized. 

Seems an extra line of code should do the trick: 

shields["US:VT:Alternate"] = banneredShield(shields["US:VT"], ["ALT"]);

However, I'm wondering if I need to also redo some data edits in OSM to remove "ALT" from `ref=*`. I think this is more in line with how other states are doing it.

<img width="528" alt="Capture" src="https://user-images.githubusercontent.com/5446730/168699511-cf671634-51f3-4e86-872d-779ce899d521.PNG">

**Comment by ZeLonewolf** [comment]
> However, I'm wondering if I need to also redo some data edits in OSM to remove "ALT" from ref=*. I think this is more in line with how other states are doing it.

Yes, I would recommend remaining consistent.  Too-long refs aren't rendered.

**Comment by zekefarwell** [comment]
Oh woops.  I had meant to add the line for alternate VT routes before #319 was merged.  We have so few bannered routes in VT that I forgot about it!  I'll add it now.  Since Americana expects refs on route relations to be 6 digits or less, `ref=122 ALT` won't be recognized as a valid ref and the shield won't show up right away.  It will show up after it's changed to `ref=122` and the next tile update happens at some point.

**Comment by ahallification** [comment]
I made the edits in the map this evening
`ref=100` `ref=122`
`modifier=Alternate` (not necessary, but the cool kids seem to be doing it)
`network=US:VT:Alternate` (unchanged in my edits, just emphasizing the tagging)

**Comment by ahallification** [comment]
Hi @zekefarwell,

I'd love to get these all done in one go, but I keep noticing additional details that were missed in my earlier reviews. Luckily, I think these are simple changes with quick wins.

**Vermont Route 7A**
`network=US:VT:Historic`
`modifier=Historic`
Action item: Add banner with "HIST"
From what I can tell, this is the first 'Historic' banner added to americana
![image](https://user-images.githubusercontent.com/5446730/171686833-7cb2de9e-bf6f-4b8e-acf5-7f80054c2373.png)

If issue https://github.com/ZeLonewolf/openstreetmap-americana/issues/357 gets any traction, than any addition of this network should comply.

**Comment by zekefarwell** [comment]
I looked into implementing the historic banner, @ahallification, but in this case I'm not sure it's necessary.  With alternate and business routes the banner needed because they are nearby a route with the same number.  For example US 30 Business is right nearby, and often intersecting with, mainline US 30.  So it's important for a map to distinguish between the two.  In this case it looks like [Vermont Route 7A](https://en.wikipedia.org/wiki/Vermont_Route_7A) is an alternate route of _US Route 7_ , so there is no other VT 7A for VT 7A Historic to be distinguished from.  I suppose we could include a HIST banner anyway just to promote that the route is historic, sort of like a scenic designation.  I don't think we've decided how to handle those yet either.
